### PR TITLE
readable results table

### PR DESCRIPTION
**Pull request by YvoneYang**
Screenshot 2024-03-12 at 7 06 02 AM

Added a table with information on selected stations and individual station KPIs on the results page

### CODE REVIEW

File: pages/results.py
Comment (thomaspaenns) [inline pages/results.py:1]: When I boot up the app I get this error:
image
Might want to change the import - I requested the change on that specific line

File: pages/results.py [hunk 1]
```diff
@@ -4,6 +4,7 @@
 from dash import dcc, html, callback, Input, Output
 import plotly.graph_objs as go
 import pandas as pd
+import dash_table
```
Comment (thomaspaenns) [inline pages/results.py:7]:
```suggestion

```
Can just be added to line 4

File: pages/results.py [hunk 2]
```diff
@@ -4,6 +4,7 @@
 from dash import dcc, html, callback, Input, Output
```
Comment (thomaspaenns) [inline pages/results.py:4]:
```suggestion
from dash import dcc, html, callback, dash_table, Input, Output
```
Import convention (should fix error)

File: pages/results.py [hunk 3]
```diff
@@ -27,14 +28,16 @@
                     title="Export data to CSV (will not export if model has not been run)"),
         ], style={'display': 'flex', 'justify-content': 'space-between'}),
         html.Div(id='overall-stats', style={"font-weight": "bold"}),
-        dcc.Graph(id='coverage-map')
+        dcc.Graph(id='coverage-map'),
```
Comment (thomaspaenns) [inline pages/results.py:31]: Can you add some whitespace between the map and the table? Perhaps a title as well (Station Stats or something)

File: pages/results.py [hunk 4]
```diff
@@ -68,7 +74,32 @@ def update_overall_stats(coverage_data, util_data, wait_time):
         csv_string = export_df.to_csv(index_label='LHRS_num')
         csv_base64 = base64.b64encode(csv_string.encode()).decode()
         download_link = "data:text/csv;charset=utf-8;base64," + csv_base64
-    return stat_str, download_link
+
+        df.reset_index(drop=True, inplace=True)
+        export_df.reset_index(drop=True, inplace=True)
+
+        export_df = pd.concat([df, export_df], axis=1).iloc[:,[0,6,30,31,32]]
+        export_df = export_df[export_df['Util']>0]
```
Comment (thomaspaenns) [inline pages/results.py:82]: Can you drop the coverage column from the table (It's kinda pointless cause the coverage will always be at or near 100% in the segments where there is a station)

File: pages/results.py [hunk 5]
```diff
@@ -27,14 +28,16 @@
                     title="Export data to CSV (will not export if model has not been run)"),
         ], style={'display': 'flex', 'justify-content': 'space-between'}),
         html.Div(id='overall-stats', style={"font-weight": "bold"}),
-        dcc.Graph(id='coverage-map')
+        dcc.Graph(id='coverage-map'),
+        html.Div(id='export-table-container')
     ], style={'marginLeft': '20px', 'marginRight': '20px'})
```
Comment (thomaspaenns) [inline pages/results.py:33]: Can you also add some whitespace below the table? (you can just use marginTop and marginBottom within the style dictionary)

**Review by thomaspaenns — CHANGES_REQUESTED** [review]
Looks good overall, a few very small changes but should only take 10ish minutes to implement

**Review by thomaspaenns — APPROVED** [review]
LGTM!

**Review by z22shaik — APPROVED** [review]
LGTM

**Pull request merged**